### PR TITLE
More assets lint fixes

### PR DESCRIPTION
**Pull request by XhmikosR**
I'm not sure about all these. I mean for one I don't think `for...in` should be used in all these cases. Which is also an error of `no-restricted-syntax`.

Most of the strict equality checks should be fine, but let's triple check

### CODE REVIEW

**Comment by emedvedev** [comment]
Authorized Now so that the changes could be tested live. So far so good :)

File: assets/badge.js [hunk 1]
```diff
@@ -75,8 +75,8 @@
 
       // update width
       var wp = 'slackin-width:' + id + ':'
-      if (wp === e.data.substr(0, wp.length)) {
-        var width = e.data.substr(wp.length)
+      if (wp === e.data.slice(0, wp.length)) {
```
Comment (XhmikosR) [inline assets/badge.js]: @emedvedev can you confirm the `slice` changes are OK? I'm going through the diff after 1.1.11

Comment (emedvedev) [reply assets/badge.js]: Should be the same in this case, yes.

Comment (XhmikosR) [reply assets/badge.js]: I meant all of the `substr` to `slice` changes, just to be safe :)

**Pull request merged**